README.md

```diff
@@ -1,6 +1,142 @@
 # Option
 
-Option is a C# representation of the maybe monad
+Option Library is a C# representation of the [maybe monad](https://en.wikibooks.org/wiki/Haskell/Understanding_monads/Maybe) which provides a good way to deal with the exceptional cases by reducing the number of  [exception and null checkings](https://en.wikipedia.org/wiki/Exception_handling#Checked_exceptions) as well as the nested blocks in the code which leads to a more compact and more readable code.
+
+### nested if statements : ugly code and hard to follow, known as [arrow anti pattern](http://wiki.c2.com/?ArrowAntiPattern)
+
+```cs
+var player = db.Players.FirstOrDefault(p => p.Name == keyword);
+if (player != null)
+{
+    Console.WriteLine(player.Name);
+    var team = db.Teams.Find(t => t.Id == player.Id);
+    if (team != null)
+    {
+        Console.WriteLine(team.Name);
+    }
+}
+```
+### can be refactored using [Guard Clauses](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) , but still with null checking
+
+```cs
+var player = db.Players.FirstOrDefault(p => p.Name == keyword);
+if (player == null)
+    return;
+Console.WriteLine(player.Name);
+var team = db.Teams.Find(t => t.Id == player.Id);
+if (team == null)
+    return;
+Console.WriteLine(team.Name);
+```
+
+### refactored using  Option Library, compact code which is easy to follow with safe chaining of steps 
+
+```cs
+Option
```

Comment:
This is much better.  But would it be possible to make it even clearer that the first two code snippets are _not_ a result from using our library.

Something like Before (in bold and red) and After (in bold and green).

Or say "Do this" followed by "Instead of this".